src/OpalCompiler-Core/OCLiteralList.class.st

```diff
@@ -1,46 +1,63 @@
 "
-Holds a unique ordered collection of literals
+Holds a unique ordered collection of literals.
+I uses an special dictionary inside that compares literals using #literalEquals message.
+I am optimised to guarantee that literals all unique and have an unique index.
 "
 Class {
 	#name : #OCLiteralList,
```

Comment:
I didn't like to do that, because I want not to extend the Dictionary for implementing OCLiteralList. Before OCLiteralList was a subclass of OrderedCollection, so it has all the methods of OrderedCollection. However, from the intended use not all the API is valid. I am extending the Dictionary just to change the comparison (I could have used PluggableDictionary, but I got less performance with the use of blocks), but I try to keep it hidden from the clients of OCLiteralList.  To me, it is more important to have a clear API for the users. 

For example, the #add: method in Collections returns the added object, but I need to return the index of the newly added literal, if you see the old code it was performing two searches in the collection and this knowledge and complexity was pushed to the client of OCLiteralList, just to keep the compatibility with Collections. Also, breaking the compatibility with Collection is bad, so I decided to separate it from the Collection hierarchy.